api/lib/domain/usecases/authenticate-pole-emploi-user.js

```diff
@@ -38,31 +38,28 @@ module.exports = async function authenticatePoleEmploiUser({
   let pixAccessToken;
 
   if (authenticatedUserId) {
-    const authenticationMethod = await authenticationMethodRepository.findOneByUserIdAndIdentityProvider({ userId: authenticatedUserId, identityProvider: AuthenticationMethod.identityProviders.POLE_EMPLOI });
-
-    if (authenticationMethod) {
-      if (authenticationMethod.externalIdentifier !== userInfo.externalIdentityId) {
-        throw new UnexpectedUserAccountError({ message: 'Le compte Pix connecté n\'est pas celui qui est attendu.' });
-      }
-
-      await authenticationMethodRepository.updatePoleEmploiAuthenticationComplementByUserId({ authenticationComplement, userId: authenticatedUserId });
-
-    } else {
-      const authenticationMethod = _buildPoleEmploiAuthenticationMethod({ userInfo, authenticationComplement, userId: authenticatedUserId });
-      await authenticationMethodRepository.create({ authenticationMethod });
-    }
-
-    pixAccessToken = tokenService.createAccessTokenFromUser(authenticatedUserId, 'pole_emploi_connect');
-
+    pixAccessToken = await _getPixAccessTokenFromAlreadyAuthenticatedPixUser({
+      userInfo,
+      authenticatedUserId,
+      authenticationComplement,
+      authenticationMethodRepository,
+      userRepository,
+      tokenService,
+    });
   } else {
     const user = await userRepository.findByPoleEmploiExternalIdentifier(userInfo.externalIdentityId);
 
     if (!user) {
       const authenticationKey = await poleEmploiTokensRepository.save(poleEmploiTokens);
-      return { authenticationKey };
+      return { authenticationKey }; // todo : refacto, should not return differents objects
```

Comment:
```suggestion
      return { authenticationKey }; // todo : refacto, should not return different objects
```

Quel est le soucis ici ?

Reply:
Le soucis que je trouve : 
- avoir plusieurs returns (ligne 54 et 66)
- ne pas voir le même objet en retour (un coup on a `{ authenticationKey }` et un autre coup on a `{ pixAccessToken, poleEmploiTokens }`
- ne pas avoir d'instance de modèle en retour et avoir des "objets js pur"